### PR TITLE
Add authorization via central OPA instance

### PR DESCRIPTION
**Pull request by tpoliaw**

### CODE REVIEW

**Review by callumforrester — DISMISSED** [review]
This makes sense to me, it would be good if we could add tests and/or prescribe a way to test it offline

File: src/graphql.rs [hunk 1]
```diff
@@ -312,6 +339,14 @@ impl Mutation {
...
         config: ConfigurationUpdates,
     ) -> async_graphql::Result<BeamlineConfiguration> {
+        if let Some(policy) = ctx.data::<Option<PolicyCheck>>()? {
+            trace!("Auth enabled: checking token");
+            let token = ctx.data::<Authorization<Bearer>>().ok();
+            policy
+                .check_admin(token, &beamline)
+                .await
+                .inspect_err(|e| info!("Authorization failed: {e:?}"))?;
+        }
```
[the remark this answers is not in the document]
Comment (tpoliaw) [reply src/graphql.rs]: Somehow extracting the function added as many lines as it saved but probably better to be in one place.

**Comment by callumforrester** [comment]
@tpoliaw I suggest we get this merged before #13 and then I'll adjust that

**Comment by tpoliaw** [comment]
> @tpoliaw I suggest we get this merged before #13 and then I'll adjust that

Sure, I'm still scattering tests over it but it should be good to go in soon

File: src/graphql/auth.rs
Comment (tpoliaw) [inline src/graphql/auth.rs:1]: Needs copyright header

Comment (stan-dot) [reply src/graphql/auth.rs:1]: https://crates.io/crates/licensure ?

Comment (tpoliaw) [reply src/graphql/auth.rs:1]: Interesting project. If there were more than 8 files to manage here I'd think about setting it up.

**Review by callumforrester — APPROVED** [review]
I'm happy I understand these changes although my understanding of OPA et al. is still limited

**Pull request merged**